spec/OpenLineage.json

```diff
@@ -0,0 +1,425 @@
+      ]
+    },
+
+    "NominalTimeRunFacet": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseFacet"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "nominalStartTime": {
+              "description": "An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal start time (included) of the run. AKA the schedule time",
+              "type": "string",
+              "format": "date-time",
+              "example": "2020-12-17T03:00:00.000Z"
+            },
+            "nominalEndTime": {
+              "description": "An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal end time (excluded) of the run. (Should be the nominal start time of the next run)",
+              "type": "string",
+              "format": "date-time",
+              "example": "2020-12-17T04:00:00.000Z"
+            }
+          },
+          "required": [
+            "nominalStartTime"
+          ]
+        }
+      ],
+      "type": "object"
+    },
+    "ParentRunFacet": {
+      "description": "the id of the parent run and job, iff this run was spawn from an other run (for example, the Dag run scheduling its tasks)",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseFacet"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "run": {
+              "type": "object",
+              "properties": {
+                "runId": {
+                  "type": "string",
+                  "format": "uuid"
+                }
+              },
+              "required": [
+                "runId"
+              ]
+            },
+            "job": {
+              "type": "object",
+              "properties": {
+                "namespace": {
+                  "description": "The namespace containing that job",
+                  "type": "string",
+                  "example": "my-scheduler-namespace"
+                },
+                "name": {
+                  "description": "The unique name for that job within that namespace",
+                  "type": "string",
+                  "example": "myjob.mytask"
+                }
+              },
+              "required": [
+                "namespace",
+                "name"
+              ]
+            }
+          },
+          "required": [
+            "run",
+            "job"
+          ]
+        }
+      ],
+      "type": "object"
+    },
+
+    "DocumentationJobFacet": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseFacet"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "description": {
+              "description": "The description of the job.",
+              "type": "string"
+            }
+          },
+          "required": [
+            "description"
+          ]
+        }
+      ],
+      "type": "object"
+    },
+    "SourceCodeLocationJobFacet": {
```

Comment:
should these canned facets also extend additional properties for future backwards compatibility for changing facet versions?

Reply:
that's a good point. Is it required?
Should all nested object types do the same?